File: Container With Most Water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int answer = 0;
        int i = 0, j = height.size() - 1;
        
        while(i < j){
            int area = min(height[i], height[j]) * (j - i);
            
            answer = max(answer, area);
            
            if(height[i] < height[j]){
                i++;
            }
            else{
                j--;
            }
        }
        
        return answer;
    }
};

File: Container With Most Water.js
/**
 * @param {number[]} height
 * @return {number}
 */
const maxArea = function(height) {
    let front = 0;
    let back = height.length - 1;
    let answer = 0;
    
    while(front < back){
        let width = 0;
        
        if(height[front] < height[back]){
            width = height[front] * (back - front);
            front++;
        } else{
            width = height[back] * (back - front);
            back--;
        }
        
        answer = answer < width ? width : answer;
    }
    
    return answer;
};